skimage/restoration/unwrap_2d_ljmu.c

```diff
@@ -138,15 +139,23 @@ EDGE *partition(EDGE *left, EDGE *right, double pivot) {
   return left;
 }
 
-void quicker_sort(EDGE *left, EDGE *right) {
+int quicker_sort(EDGE *left, EDGE *right) {
   EDGE *p;
   double pivot;
+  int error_code = 0;
 
   if (find_pivot(left, right, &pivot) == yes) {
+    if (isnan(pivot)) {
+      return NAN_ERROR;
+    }
```

Comment:
Not sure if that is the best way to address the problem. I understand little of what's going on, but it's better than hanging with one core stuck at 100%. Performance wise I didn't see any significant impact for the following snippet taken from plot_phase_unwrap.py:

```python
import numpy as np
from skimage import data, img_as_float, color, exposure
from skimage.restoration import unwrap_phase
image = color.rgb2gray(img_as_float(data.chelsea()))
image = exposure.rescale_intensity(image, out_range=(0, 4 * np.pi))
image_wrapped = np.angle(np.exp(1j * image))

%timeit unwrap_phase(image_wrapped)
```

Though I am a bit dubious because the [paper](https://www.academia.edu/download/50543963/AO.41.00743720161125-19313-xgdwf3.pdf) makes it seem like there should be  a huge amount of edges and I am not sure how often `quicker_sort` is called in the total scheme of things.